### DISCUSSION TITLE
difference between Alpine 2 and 3 when comes to nested x-for under x-if?

### DISCUSSION THREAD

**Discussion by sqllyw** [alpinejs/alpine#1444] [7. Bugs (archived)] [answered]
Hi,
I was trying to upgrade to Alphine 3, encounter a issue, here is a very small demo:

https://jsfiddle.net/sqj7oLxm/

above uses Alpine 2, it has first x-for showing names, under every name another x-for shows colors, this x-for is under a x-if:

     <template x-if="!hideColor">

so in the first loading of page, it will show name and color as hideColor is false, then when you click the button to set hideColor to true, color will not be shown. 

here is the exact code using alpine 3, 

https://jsfiddle.net/sqj7oLxm/1/

in the first loading of page, color is not shown, after clicking to set hideColor to true, it start to work the same as Alpine 2 version, could anybody take a look at examples and point out why  <template x-if="!hideColor"> is not working in first loading of page?

Thanks

**Comment by KevinBatdorf** [comment]
That seems like a bug. Here's a more minimal foo/bar type test case to use.

https://codepen.io/KevinBatdorf/pen/LYWgKZb?editors=1010

I'll attempt to debug this tomorrow but [the first line of the loop ](https://github.com/alpinejs/alpine/blob/b46c41fa240cd8af2dcaa29fb60fb1db0389c95a/packages/alpinejs/src/directives/x-for.js#L38)makes me think it will take a lot more experience :)

> // Prepare yourself. There's a lot going on here.

**Reply by sqllyw** [reply]
thanks for confirming this as a bug, I have moved this to 'Bug' category

**Comment by sqllyw** [comment]
seems that you can not use a x-if before the second x-for, a workaround is:

https://codepen.io/sqllyw/pen/KKWrRjW

not a very good solution, but needed to proceed with the upgrading of project to Alpine 3

**Comment by sqllyw** [comment, marked as answer]
tested this in 3.1.0, and it's working:

https://codepen.io/KevinBatdorf/pen/LYWgKZb?editors=1010
